cli/azd/pkg/auth/token.go

```diff
@@ -18,61 +18,94 @@ var jwtClaimsRegex = regexp.MustCompile(`^[a-zA-Z0-9-_]*\.([a-zA-Z0-9-_]*)\.[a-z
 
 // cspell: enable
 
-func GetTenantIdFromToken(token string) (string, error) {
-	return getTidClaimFromAccessToken(token)
+// TokenClaims contains claims about a user from an access token.
+// https://learn.microsoft.com/en-us/entra/identity-platform/access-token-claims-reference.
+type TokenClaims struct {
+	PreferredUsername string `json:"preferred_username,omitempty"`
```

Comment:
Would you want these to be `*string` and `*int64` to denote that these claims may not always be present?  I'm not sure what, if any, claims STS will always staple to an access token, but I suspect some of these (FamilyName, MiddleName, etc) may not exist when using an SP?

Reply:
In practice, most of these claims end up being *optional* in some form -- whether it's 1.0 vs. 2.0, or whether specific claims were included in the `/authorize` request. Ergonomically speaking, it feels that pointers ends up being worse. FWIW, this is what MSAL go did for modeling the token claims and it works for them.